kafka/producer.go

```
@@ -461,6 +464,27 @@ func NewProducer(conf *ConfigMap) (*Producer, error) {
 	}
 	p.handle.fwdDr = v.(bool)
 
+	v, err = confCopy.extract("go.delivery.report.fields", "all")
+	if err != nil {
+		return nil, err
+	}
+	p.handle.mf = newMessageFields()
+	switch v {
+	case "all":
+		// nothing to do
+	default:
+		// The "none" case is implicit here
+		p.handle.mf.disableAll()
+		for _, value := range strings.Split(v.(string), ",") {
+			switch value {
+			case "key":
+				p.handle.mf.key = true
+			case "value":
+				p.handle.mf.value = true
+			}
```

Should error out on unknown fields.